From c962aa40e970b002dd48e908194235ff7fd299c0 Mon Sep 17 00:00:00 2001
From: sonan <sonan@ami.com>
Date: Wed, 20 Mar 2024 10:26:55 +0530
Subject: [PATCH 52/52] Add Restriction Mode Interface

Add Restriction mode interface to settings D-Bus object.

Tested in AST2600-EVB:
-Verified D-Bus property is exposed as expected

root@intel-obmc:~# busctl introspect xyz.openbmc_project.Settings /xyz/openbmc_project/control/host0/restriction_mode
NAME                                                 TYPE      SIGNATURE RESULT/VALUE                             FLAGS
org.freedesktop.DBus.Introspectable                  interface -         -                                        -
.Introspect                                          method    -         s                                        -
org.freedesktop.DBus.Peer                            interface -         -                                        -
.GetMachineId                                        method    -         s                                        -
.Ping                                                method    -         -                                        -
org.freedesktop.DBus.Properties                      interface -         -                                        -
.Get                                                 method    ss        v                                        -
.GetAll                                              method    s         a{sv}                                    -
.Set                                                 method    ssv       -                                        -
.PropertiesChanged                                   signal    sa{sv}as  -                                        -
xyz.openbmc_project.Control.Security.RestrictionMode interface -         -                                        -
.RestrictionMode                                     property  s         "xyz.openbmc_project.Control.Security... emits-chane
root@intel-obmc:~#

root@intel-obmc:~# busctl get-property xyz.openbmc_project.Settings /xyz/openbmc_project/control/host0/restriction_mode xyz.o
penbmc_project.Control.Security.RestrictionMode RestrictionMode
s "xyz.openbmc_project.Control.Security.RestrictionMode.Modes.Provisioning"
root@intel-obmc:~#

Signed-off-by: sonan <sonan@ami.com>
---
 include/defaults.hpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/include/defaults.hpp b/include/defaults.hpp
index f1b258e..ad18f92 100644
--- a/include/defaults.hpp
+++ b/include/defaults.hpp
@@ -91,6 +91,14 @@ inline void loadSettings(sdbusplus::asio::object_server &objectServer,
 
     setting->addProperty("PowerRestoreDelay", static_cast<uint64_t>(0));
 
+    setting = &settings.emplace_back(
+        objectServer, "/xyz/openbmc_project/control/host0/restriction_mode",
+        "xyz.openbmc_project.Control.Security.RestrictionMode");
+
+    setting->addProperty("RestrictionMode",
+                         "xyz.openbmc_project.Control.Security.RestrictionMode."
+                         "Modes.ProvisionedHostDisabled");
+
     setting = &settings.emplace_back(
         objectServer, "/xyz/openbmc_project/control/host0/acpi_power_state",
         "xyz.openbmc_project.Control.Power.ACPIPowerState");
-- 
2.43.0

